salt/modules/boto3_elasticsearch.py

```diff
@@ -156,7 +150,7 @@ def add_tags(
         boto_params = {
             "ARN": arn,
             "TagList": [
-                {"Key": k, "Value": value} for k, value in six.iteritems(tags or {})
+                {"Key": k, "Value": value} for k, value in tags.items() or {}.items()
```

Comment:
```suggestion
	              {"Key": k, "Value": value} for k, value in (tags or {}).items()
```
Yeah, that pre-commit horked this line up completely. Something seems off with the indention here? but... this is the logic fix.

Reply:
cc @s0undt3ch

Reply:
I think what's in the PR right now is logically ok, but yours is clearer, I'll add that.

What the pre-commit did originally was change `six.iteritems(tags or {})` to `tags or {}.items()`, which is definitely not ok!

Reply:
what's in the PR now will fail when `tags` is None: 

```
>>> tags = None
>>> tags.items() or {}.items()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'items'
```

That was an implicit fallback with `iteritems(tags or {})` 🙃 

The other option would be `tags.items() if tags else {}.items()` but that would be silly 😂

Reply:
:facepalm:

Reply:
I am really confused now.  How did this manage to get in to 3002 in d089e810fe/https://github.com/saltstack/salt/pull/58622?  The reason I saw and fixed this (not quite correctly) in the first place is that it was failing a test.  How did that not happen for the 3002 release?

Reply:
Ah, I see, the comprehension was changed: https://github.com/saltstack/salt/pull/58622/files#diff-fe8071ebcb4727856197b666dbf829930088a5d5fc5d8c82a0d806d03ecf059bR149